src/main.ts

```diff
@@ -87,10 +87,10 @@ export const loop = ErrorMapper.wrapLoop(() => {
       if (sources.length > 0) {
 
         // We can farm! We'll choose the first one.
-        let source = sources[0];
+        let source = mycreep.pos.findClosestByRange(FIND_SOURCES);
 
-        // if creep has no energy, go to the energy source and harvest some
-        if(mycreep.store[RESOURCE_ENERGY] == 0) {
+        // if creep has energy less than or equal to 48, go to the energy source and harvest some
+        if(mycreep.store[RESOURCE_ENERGY] <= 48) {
```

Comment:
So the creep will now harvest to 48, then travel to the controller, and dump 1 energy in to the controller.  This would bring it back below 48 and it will return to the source will a full load.

Reply:
```
// While you're harvesting continue until you're full.
if (mycreep.memory.objective == true) {
  // Goto and harvest source

  if (mycreep.store[RESOURCE_ENERGY] < mycreep.store.getCapacity()) {
    mycreep.memory.objective = false
  }
} else {
  // Goto and upgrade controller

  if (mycreep.store[RESOURCE_ENERGY] < mycreep.store.getCapacity()) {
    mycreep.memory.isHarvesting = true;
  }
}
```